project/src/js/ui/tournament/list/tournamentListView.js

```diff
@@ -0,0 +1,76 @@
+import h from '../../helper';
+import {header, empty, pad} from '../../shared/common';
+import layout from '../../layout';
+import i18n from '../../../i18n';
+import m from 'mithril';
+import tabs from 'polythene/tabs/tabs';
+
+export default function view(ctrl) {
+  const bodyCtrl = tournamentListBody.bind(undefined, ctrl);
+
+  return layout.free(header.bind(undefined, i18n('tournaments')), bodyCtrl, empty, empty);
+}
+
+const TABS = [{
+    id: 'started',
+    label: 'In Progress'
+}, {
+    id: 'created',
+    label: 'Upcoming'
+}, {
+    id: 'finished',
+    label: 'Completed'
+}];
+
+function tabNavigation (currentTabFn) {
+    return m('.nav-header', m.component(tabs, {
+        buttons: TABS,
+        autofit: true,
+        selectedTab: currentTabFn(),
+        activeSelected: true,
+        getState: (state) => {
+            currentTabFn(state.index);
+        }
+    }));
+}
+
+function tournamentListBody(ctrl) {
+  if (!ctrl.tournaments()) return null;
```

Comment:
Done. I did the same for tournamentView as well.

On Fri, Feb 19, 2016 at 8:42 AM, Vincent Velociter <notifications@github.com
> wrote:

> In project/src/js/ui/tournament/list/tournamentListView.js
> <https://github.com/veloce/lichobile/pull/281#discussion_r53461327>:
>
> > +}];
> > +
> > +function tabNavigation (currentTabFn) {
> > +    return m('.nav-header', m.component(tabs, {
> > +        buttons: TABS,
> > +        autofit: true,
> > +        selectedTab: currentTabFn(),
> > +        activeSelected: true,
> > +        getState: (state) => {
> > +            currentTabFn(state.index);
> > +        }
> > +    }));
> > +}
> > +
> > +function tournamentListBody(ctrl) {
> > +  if (!ctrl.tournaments()) return null;
>
> I see.
> The problem is you declared tournaments like so:
>
> const tournaments = m.prop([]);
>
> Until the promise is resolved, ctrl.tournaments() will return [], which
> evaluates to true;
>
> To fis this you need to declare tournaments like:
>
> const tournaments = m.prop({});
>
> And then you need to check if object is empty, using lodash:
>
> if (isEmpty(ctrl.tournaments())) return null;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/veloce/lichobile/pull/281/files#r53461327>.
>